src/main/java/uk/gov/justice/probation/courtcaseservice/service/mapper/CourtCaseMapper.java

```diff
@@ -32,16 +41,129 @@ public static CourtCaseEntity create(CourtCaseEntity courtCaseEntity, String upd
+                            .map(CourtCaseMapper::createHearing)
+                            .collect(Collectors.toList()))
             .offences(courtCaseEntity.getOffences()
                             .stream()
                             .map(CourtCaseMapper::createOffence)
                             .collect(Collectors.toList()))
+            .defendants(Optional.ofNullable(courtCaseEntity.getDefendants()).orElse(Collections.emptyList())
+                .stream()
+                .map(defendant -> createDefendant(defendant, crn, updatedProbationStatus))
+                .collect(Collectors.toList()))
             .build();
 
         newCourtCaseEntity.getOffences().forEach(offenceEntity -> offenceEntity.setCourtCase(newCourtCaseEntity));
+        newCourtCaseEntity.getDefendants().forEach(defendantEntity -> defendantEntity.setCourtCase(courtCaseEntity));
+        newCourtCaseEntity.getHearings().forEach(hearingEntity -> hearingEntity.setCourtCase(courtCaseEntity));
         return newCourtCaseEntity;
     }
 
+    public static CourtCaseEntity mergeDefendantsOnCase(CourtCaseEntity existingCase, CourtCaseEntity updatedCase, String defendantId) {
+        final var existingDefendants = Optional.ofNullable(existingCase.getDefendants()).orElse(Collections.emptyList());
+        final var existingHearings = CourtCaseMapper.createHearings(existingCase.getHearings());
+        if (existingDefendants.size() <= 1) {
+            final var caseToSave = updatedCase.withHearings(existingHearings);
+            applyParentToCollections(caseToSave);
+            return caseToSave;
+        }
+
+        final var updatedDefendant = updatedCase.getDefendants().stream()
+            .filter(existingDef -> defendantId.equalsIgnoreCase(existingDef.getDefendantId()))
+            .findFirst()
+            .orElseThrow(() -> new ConflictingInputException(String.format("No defendant ID matching %s in updated case", defendantId)));
+
+        // Clone all OTHER defendants, removing the IDs
+        final var allDefendants = existingDefendants.stream()
+            .filter(existingDef -> !defendantId.equalsIgnoreCase(existingDef.getDefendantId()))
+            .map(otherDefendant -> CourtCaseMapper.createDefendant(otherDefendant, null))
+            .collect(Collectors.toCollection(() -> new ArrayList<>(Arrays.asList(updatedDefendant))));
+
+        // rebuild the case with new defendants and hearings
+        final var caseToSave = updatedCase.withDefendants(allDefendants)
+                                                        .withHearings(CourtCaseMapper.createHearings(existingCase.getHearings()));
+        applyParentToCollections(caseToSave);
+        return caseToSave;
+    }
+
+    private static void applyParentToCollections(final CourtCaseEntity caseToSave) {
+        Optional.ofNullable(caseToSave.getDefendants()).orElse(Collections.emptyList())
+            .forEach(defendantEntity -> defendantEntity.setCourtCase(caseToSave));
```

Comment:
With the JPA entities, they haven't been immutable since we made them. The problem has been the bi-directional links when we create them. Our mappers ave always tended to build the parent, then teh members and then apply parentage. Happy to try "with"  in this case but there are still setters.